protege-launcher/src/main/java/org/protege/prefs/FilePrefs.java

```diff
@@ -0,0 +1,192 @@
+	    }
+	  }
+	 
+	  synchronized protected String getSpi(String key) {
+	    return root.get(key);
+	  }
+	 
+	  synchronized protected void removeSpi(String key) {
+	    root.remove(key);
+	    try {
+	      flush();
+	    }
+	    catch (BackingStoreException e) {
+	      log.error("Unable to flush after removing " + key, e);
+	    }
+	  }
+	 
+	  synchronized protected void removeNodeSpi() throws BackingStoreException {
+	    isRemoved = true;
+	    flush();
+	  }
+	 
+	  synchronized protected String[] keysSpi() throws BackingStoreException {
+	    return root.keySet().toArray(new String[root.keySet().size()]);
+	  }
+	 
+	  synchronized protected String[] childrenNamesSpi() throws BackingStoreException {
+		  return children_names.toArray(new String[children_names.size()]);
+	  }
+	 
+	  synchronized protected FilePrefs childSpi(String name) {
+	    FilePrefs child = children.get(name);
+	    if (child == null || child.isRemoved()) {
+	      child = new FilePrefs(this, name);
+	      children.put(name, child);
+	    }
+	    return child;
+	  }
+	 
+	 
+	  protected synchronized void syncSpi() throws BackingStoreException {
+		  
+		  if (isRemoved()) return;
+
+		  final File file = FileBackingStorePrefsFactory.getPreferencesFile();
+
+		  if (!file.exists()) return;
+
+		  Properties p = new Properties();
+		  try {
+			  p.load(new FileInputStream(file));
+
+			  StringBuilder sb = new StringBuilder();
+			  getPath(sb);
+			  String path = sb.toString();
+
+			  final Enumeration<?> pnen = p.propertyNames();
+			  while (pnen.hasMoreElements()) {
+				  String propKey = (String) pnen.nextElement();
+				  if (propKey.startsWith(path)) {
+					  String subKey = propKey.substring(path.length());
+					  // Only load immediate descendents
+					  if (subKey.indexOf('/') == -1) {
+						  root.put(subKey, p.getProperty(propKey));
+					  } else {
+						  children_names.add(subKey.substring(0,subKey.indexOf('/')));
+					  }
+				  }
+			  }
+		  }
+		  catch (IOException e) {
+			  throw new BackingStoreException(e);
+		  }
+	  }
+	  
+	 
+	  private void getPath(StringBuilder sb) {
+	    final FilePrefs parent = (FilePrefs) parent();
+	    if (parent == null) return;
+	 
+	    parent.getPath(sb);
+	    sb.append(name()).append('/');
+	  }
+	 
+	  protected synchronized void flushSpi() throws BackingStoreException {
+		  
+		  final File file = FileBackingStorePrefsFactory.getPreferencesFile();
+		  
+		  try {
+
+			  StringBuilder sb = new StringBuilder();
+			  getPath(sb);
+			  String path = sb.toString();
+			  Properties p = new Properties();
+			  if (file.exists()) {
+				  
+				  p.load(new FileInputStream(file));
+
+				  List<String> toRemove = new ArrayList<String>();
+
+				  // Make a list of all direct children of this node to be removed
+				  final Enumeration<?> pnen = p.propertyNames();
+				  while (pnen.hasMoreElements()) {
+					  String propKey = (String) pnen.nextElement();
+					  if (propKey.startsWith(path)) {
```

Comment:
ok, refactored this, it's one of those case where the function does so little one ends up with more lines of code.